File: backEnd/src/main/java/com/quiz/ourclass/domain/board/entity/Comment.java
package com.quiz.ourclass.domain.board.entity;

import com.quiz.ourclass.domain.member.entity.Member;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import java.time.LocalDateTime;

@Entity
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id;
    @ManyToOne(fetch = FetchType.LAZY)
    Member member;
    String content;
    LocalDateTime createTime;
    LocalDateTime updateTime;
    Long parentId;
}

File: android/core/model/src/main/java/com/sixkids/model/Comment.kt
package com.sixkids.model

data class Comment(
    val id: Long,
    val member: MemberSimple,
    val content: String,
    val createTime: String,
    val updateTime: String,
    val recomments: List<Recomment>,
)

data class Recomment(
    val id: Long,
    val member: MemberSimple,
    val content: String,
    val createTime: String,
    val updateTime: String,
    val parentId: Long,
)
